### ISSUE TITLE
linter: no-useless-return false positive on early return after statement in switch case

### ISSUE THREAD

**Issue by groomain** [oxc-project/oxc#23978]
### What version of Oxlint are you using?

1.71.0

### What command did you run?

oxlint -c 

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "rules": {
    "no-useless-return": "error"
  }
}
```

### What happened?

Oxlint reports no-useless-return on early return statements that are not useless — they prevent subsequent code in the same switch case from running.

[Steps to reproduce](https://playground.oxc.rs/?lintRules=eslint%2Fno-useless-return&options=%7B%22run%22%3A%7B%22lint%22%3Atrue%2C%22formatter%22%3Afalse%2C%22transform%22%3Afalse%2C%22isolatedDeclarations%22%3Afalse%2C%22whitespace%22%3Afalse%2C%22mangle%22%3Afalse%2C%22compress%22%3Afalse%2C%22scope%22%3Atrue%2C%22symbol%22%3Atrue%2C%22cfg%22%3Afalse%7D%2C%22parser%22%3A%7B%22extension%22%3A%22tsx%22%2C%22allowReturnOutsideFunction%22%3Atrue%2C%22preserveParens%22%3Atrue%2C%22allowV8Intrinsics%22%3Atrue%2C%22semanticErrors%22%3Atrue%7D%2C%22linter%22%3A%7B%22config%22%3A%22%7B%5C%22categories%5C%22%3A%7B%5C%22correctness%5C%22%3A%5C%22off%5C%22%7D%2C%5C%22rules%5C%22%3A%7B%5C%22eslint%2Fno-useless-return%5C%22%3A%5C%22error%5C%22%7D%7D%22%7D%2C%22formatter%22%3A%7B%22useTabs%22%3Afalse%2C%22tabWidth%22%3A2%2C%22endOfLine%22%3A%22lf%22%2C%22printWidth%22%3A80%2C%22singleQuote%22%3Afalse%2C%22jsxSingleQuote%22%3Afalse%2C%22quoteProps%22%3A%22as-needed%22%2C%22trailingComma%22%3A%22all%22%2C%22semi%22%3Atrue%2C%22arrowParens%22%3A%22always%22%2C%22bracketSpacing%22%3Atrue%2C%22bracketSameLine%22%3Afalse%2C%22objectWrap%22%3A%22preserve%22%2C%22singleAttributePerLine%22%3Afalse%7D%2C%22transformer%22%3A%7B%22target%22%3A%22es2015%22%2C%22useDefineForClassFields%22%3Atrue%2C%22experimentalDecorators%22%3Atrue%2C%22emitDecoratorMetadata%22%3Atrue%2C%22optimizeEnums%22%3Atrue%2C%22optimizeConstEnums%22%3Atrue%7D%2C%22isolatedDeclarations%22%3A%7B%22stripInternal%22%3Afalse%7D%2C%22codegen%22%3A%7B%22normal%22%3Atrue%2C%22jsdoc%22%3Atrue%2C%22annotation%22%3Atrue%2C%22legal%22%3Atrue%7D%2C%22compress%22%3A%7B%7D%2C%22mangle%22%3A%7B%22topLevel%22%3Atrue%2C%22keepNames%22%3Afalse%7D%2C%22controlFlow%22%3A%7B%22verbose%22%3Afalse%7D%2C%22inject%22%3A%7B%22inject%22%3A%7B%7D%7D%2C%22define%22%3A%7B%22define%22%3A%7B%7D%7D%7D&code=import+React%2C+%7B+useEffect%2C+useRef+%7D+from+%27react%27%0A%0A%0Aconst+changeStep+%3D+%28%29+%3D%3E+%7B%0A++switch+%28direction%29+%7B%0A++++case+DIRECTION.BACKWARD%3A%0A++++++if+%28step+%3D%3D%3D+STEPS.Step1%29+%7B%0A++++++++setIsFlowShown%28false%29%0A++++++++return%0A++++++%7D%0A++++++setStep%281%29%0A++++++break%0A++++case+DIRECTION.FORWARD%3A%0A++++++if+%28step+%3D%3D%3D+STEPS.Step5%29+%7B%0A++++++++setIsFlowShown%28false%29%0A++++++++return%0A++++++%7D%0A++++++setStep%282%29%0A++++++break%0A++%7D%0A%7D%0A%0Aexport+default+DummyComponent)
